src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -1333,6 +1224,94 @@ def construct_payload(self):
         self.set_up_service_binds()
         self.set_up_extended_location()
 
+        if self.get_argument_source():
+            self._set_up_source_or_repo()
+
+        if self.get_argument_repo():
+            container_app = self._get_containerapp_if_exists()
+            if container_app:
+                # Executing the create command with --repo makes two ARM calls. The container app is created first
+                # and the source control is set only after the container app is created. This means, when the container app is
+                # first created, the image would be defaulted to mcr.microsoft.com/k8se/quickstart:latest (if --image was not provided)
+                # and the image name would change to `registry-server/containerapp-name:{github-sha}` only
+                # after the source control completes successfully
+                # and the new image is deployed.
+                # If the create command is executed multiple times however, we need to make sure that the image is not defaulted
+                # to mcr.microsoft.com/k8se/quickstart:latest if source control fails the subsequent times by setting the image
+                # to the previously deployed image.
+
+                if container_app["properties"]["template"]["containers"] is None or len(container_app["properties"]["template"]["containers"]) == 0:
+                    raise ValidationError(
+                        "The container app '{}' does not have any containers. Please use --image to set the image for the container app".format(
+                            self.get_argument_name()))
+                if container_app["properties"]["template"]["containers"][0]["image"] is None or container_app["properties"]["template"]["containers"][0]["image"] == HELLO_WORLD_IMAGE:
+                    raise ValidationError(
+                        "The container app '{}' does not have an image or is mcr.microsoft.com/k8se/quickstart:latest. Please use --image to set the image for the container app while creating the container app.".format(
+                            self.get_argument_name()))
+                if self.containerapp_def["properties"]["template"]["containers"] is None or len(self.containerapp_def["properties"]["template"]["containers"]) == 0:
+                    raise ValidationError(
+                        "The container app '{}' does not have any containers. Please use --image to set the image for the container app".format(
+                            self.get_argument_name()))
+                self.containerapp_def["properties"]["template"]["containers"][0]["image"] = container_app["properties"]["template"]["containers"][0]["image"]
+
+    def _get_containerapp_if_exists(self):
```

Comment:
`_get_containerapp_if_exists` is duplicate with `self.show()`.
If not 404 status code, with be handle as not Found.
Recommand to use `handle_non_404_status_code_exception`

 ```
       except Exception as e:
            handle_non_404_status_code_exception(e)
```

Reply:
For the `create` command however, if we leverage `self.show` it may always result in a 404 if run with `--repo` arg because the container app would not exist the first time.
We should update the image only if the create command was already executed before.

Reply:
OK, I see. You need a function to get containerapp if exists, if not existed, return None. 

If 5xx status code, with be handle as not Found.
Recommend to use handle_non_404_status_code_exception

      except Exception as e:
           handle_non_404_status_code_exception(e)

Reply:
Updated.